src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -106,27 +113,56 @@ private Button createSaveButton() {
         Button save = new Button();
         save.setText("Save");
         save.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
-        save.setDisableOnClick(true);
         save.addClassName("shelfFormSaveButton");
         save.addClickListener(event -> validateOnSave());
         return save;
     }
 
+    private Button createDeleteButton() {
+        delete = new Button();
+        delete.setText("Delete shelf");
+        delete.addThemeVariants(ButtonVariant.LUMO_ERROR);
+        delete.addClickListener(event -> validateOnDelete());
+        return delete;
+    }
+
     private void validateOnSave() {
         if (binder.isValid()) {
             setCustomShelfBean();
             fireEvent(new SaveEvent(this, binder.getBean()));
             closeForm();
             if (binder.getBean() != null) {
                 showSaveConfirmation();
+                owner.setEditEnabled(false);
             } else {
                 showSaveError();
             }
         }
     }
 
+    private void validateOnDelete() {
```

Comment:
May be we can add a test for this as well?

Reply:
BooksInShelfViewTest will have a deleteShelfOnDeleteButtonClick test built for this.

Reply:
I would say its better if we can write the test along with the actual code instead of leaving it for later?

Reply:
What I meant by this is that I had to recently reset my local environment so I have a new branch with this change included. My apologies for being unclear.